### PR TITLE
Fix JVMTI methods

### PR DESCRIPTION
**Pull request by babsingh**
- Return JVMTI_ERROR_INVALID_THREAD if the input thread parameter is
not a thread object.
- Consolidate j.l.Thread and j.l.VirtualThread error-checking within
getVMThread. This avoids repeated NULL checking and potentially
fetching the current thread.
- Convert getVMThread's boolean input parameters to a bit field.
- More verbose naming used to avoid ambiguity: IS_JAVA_LANG_THREAD
and IS_JAVA_LANG_VIRTUALTHREAD.

Fixes: #15986

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
Fixes serviceability/jvmti/negative/contmon03/contmon03.java in https://github.com/eclipse-openj9/openj9/issues/15986.

**Comment by babsingh** [comment]
@gacholio Can you please review these changes?

File: runtime/jvmti/jvmtiThread.c [hunk 1]
```diff
@@ -408,6 +407,8 @@ jvmtiInterruptThread(jvmtiEnv *env,
 		
 		ENSURE_PHASE_LIVE(env);
 		ENSURE_CAPABILITY(env, can_signal_thread);
+		ENSURE_JTHREAD_NON_NULL(thread);
+		ENSURE_JTHREAD(currentThread, thread);
```
Comment (gacholio) [inline runtime/jvmti/jvmtiThread.c:411]: If possible, it would be nice to combine this, as the original `ENSURE_JTHREAD_NON_NULL` is completely misleading (it does not check for thread at all, just the NULL). Alternately, replace all uses of `ENSURE_JTHREAD_NON_NULL` with `ENSURE_JOBJECT_NON_NULL` and delete the misleading one entirely.

File: runtime/jvmti/jvmtiStackFrame.c [hunk 2]
```diff
@@ -53,11 +53,14 @@ jvmtiGetStackTrace(jvmtiEnv* env,
 		vmFuncs->internalEnterVMFromJNI(currentThread);
 
 		ENSURE_PHASE_LIVE(env);
-
 		ENSURE_NON_NEGATIVE(max_frame_count);
 		ENSURE_NON_NULL(frame_buffer);
 		ENSURE_NON_NULL(count_ptr);
 
+		if (NULL != thread) {
```
Comment (gacholio) [inline runtime/jvmti/jvmtiStackFrame.c:60]: Might be nice to add a new macro for this since it comes up many times. `ENSURE_JTHREAD_ALLOW_NULL` perhaps.

File: runtime/jvmti/jvmtiThread.c [hunk 3]
```diff
@@ -408,6 +407,8 @@ jvmtiInterruptThread(jvmtiEnv *env,
 		
 		ENSURE_PHASE_LIVE(env);
 		ENSURE_CAPABILITY(env, can_signal_thread);
+		ENSURE_JTHREAD_NON_NULL(thread);
```
[the remark this answers is not in the document]
Comment (gacholio) [reply runtime/jvmti/jvmtiThread.c:410]: `getVMThread` will result in the correct error if NULL is passed and `allowNULL` is false, so this should simply be removed.

**Comment by babsingh** [comment]
getVMThread will throw JVMTI_ERROR_INVALID_THREAD if allowNull is FALSE. Moving ENSURE_JTHREAD check inside getVMThread will simplify the change. Let me give it a try.

**Comment by babsingh** [comment]
@gacholio Reworked the PR. All earlier comments have been addressed.

File: runtime/oti/jvmtiInternal.h [hunk 4]
```diff
@@ -489,7 +481,9 @@ typedef struct jvmtiGcp_translation {
 #if JAVA_SPEC_VERSION >= 19
 #define ENSURE_JTHREAD_NOT_VIRTUAL(vmThread, jthrd, error) \
 	do { \
-		if (IS_VIRTUAL_THREAD((vmThread), J9_JNI_UNWRAP_REFERENCE(jthrd))) { \
+		if ((NULL != (jthrd)) \
```
Comment (gacholio) [inline runtime/oti/jvmtiInternal.h:484]: Unless I'm misreading, this does not handle the case of a NULL input parameter where the implied current thread is virtual.

Given that you've moved the NULL check into `getVMThread`, perhaps we should just add another parameter `allowVirtual` and keep all the tests in one place to avoid repeatedly NULL checking and potentially fetching the current thread.

File: runtime/oti/j9.h [hunk 5]
```diff
@@ -359,6 +359,9 @@ static const struct { \
 #define IS_VIRTUAL_THREAD(vmThread, object) \
 	isSameOrSuperClassOf(J9VMJAVALANGBASEVIRTUALTHREAD((vmThread)->javaVM), J9OBJECT_CLAZZ(vmThread, object))
 
+#define IS_JTHREAD(vmThread, object) \
```
[the remark this answers is not in the document]
Comment (babsingh) [reply runtime/oti/j9.h]: Renamed to `IS_JAVA_LANG_THREAD`.

**Comment by babsingh** [comment]
Will have to retain ENSURE_JTHREAD macro since jvmtiRunAgentThread does not use getVMThread.

File: runtime/oti/jvmtiInternal.h (hunk 4 above)
Comment (babsingh) [reply runtime/oti/jvmtiInternal.h]: In addition to `allowVirtual` (bitfield), we will also have to pass the type of error. Different methods throw different errors vthreads: JVMTI_ERROR_UNSUPPORTED_OPERATION and JVMTI_ERROR_OPAQUE_FRAME.

File: runtime/jvmti/jvmtiExtensionMechanism.c [hunk 6]
```diff
@@ -1465,13 +1465,14 @@ jvmtiGetOSThreadID(jvmtiEnv* jvmti_env, ...)
 		ENSURE_PHASE_START_OR_LIVE(jvmti_env);
 		ENSURE_NON_NULL(threadid_ptr);
 
+		rc = getVMThread(
+				currentThread, thread, &targetThread,
 #if JAVA_SPEC_VERSION >= 19
```
Comment (gacholio) [inline runtime/jvmti/jvmtiExtensionMechanism.c:1470]: This ifdef is unnecessary as the vthread errors only happen on 19+ anyway.

Comment (babsingh) [reply runtime/jvmti/jvmtiExtensionMechanism.c:1470]: JVMTI spec has introduced `JVMTI_ERROR_UNSUPPORTED_OPERATION` in JDK19+.

File: runtime/oti/jvmtiInternal.h [hunk 7]
```diff
@@ -61,6 +61,9 @@ typedef enum {
 
 #define J9JVMTI_UDATA_BITS (sizeof(UDATA) * 8)
 
+#define J9JVMTI_GETVMTHREAD_ERROR_ON_NULL_JTHREAD 0x1
```
Comment (gacholio) [inline runtime/oti/jvmtiInternal.h:64]: These are bits so these values are incorrect.

Comment (babsingh) [reply runtime/oti/jvmtiInternal.h:64]: Yes, the last one should be 0x4.

**Review by gacholio — APPROVED** [review]

File: runtime/oti/jvmtiInternal.h [hunk 8]
```diff
@@ -61,6 +61,9 @@ typedef enum {
...
+#define J9JVMTI_GETVMTHREAD_IGNORE_NOT_ALIVE_JTHREAD 0x2
```
[the remark this answers is not in the document]
Comment (babsingh) [reply runtime/oti/jvmtiInternal.h]: GTG. All previous comments have been addressed.

**Comment by gacholio** [comment]
jenkins test sanity win jdk8

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk19
